lightning/src/onion_message/messenger.rs

```diff
@@ -1102,6 +1195,11 @@ where
 				.entry(*their_node_id)
 				.or_insert_with(|| OnionMessageRecipient::ConnectedPeer(VecDeque::new()))
 				.mark_connected();
+			if self.intercept_messages_for_offline_peers {
+				self.enqueue_event(
+					Event::OnionMessagePeerConnected { peer_node_id: *their_node_id }
+				);
```

Comment:
Seems like it could be a problem if we enqueue a bunch of `Event::OnionMessageForOfflinePeer` but end up dropping an `Event::OnionMessagePeerConnected` for the corresponding peer because of a full buffer. Maybe events for a different peer cause the buffer to fill up, for instance. Should we unconditionally enqueue the latter type of events?

Reply:
As mentioned https://github.com/lightningdevkit/rust-lightning/pull/2973#discussion_r1567982950, the events queue should never get close to filling up. So I'm not sure it's worth adding special handling for event dropping cases?